## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#1f93e891-19c1-488c-a86f-d8b068b23433) |
| mint-worm.sh | :x: [more...](#15b7f0db-63d2-45fb-9054-be609c691927) |
| mint-dist-xl.sh | :x: [more...](#431a59f9-39bf-4c6e-aba4-7973f8dcea38) |
<a name="1f93e891-19c1-488c-a86f-d8b068b23433"></a>
### 7556-f913abb/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32234
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 15ae7f124d4a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-jggpcv90zvdg6j",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 359,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:32234/aws-sdk-go-test-jggpcv90zvdg6j: dial tcp 72.28.97.59:32234: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="15b7f0db-63d2-45fb-9054-be609c691927"></a>
### 7556-f913abb/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31095
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ded3c0f0ce6b:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-05f1jl9fqagrykcwo2uaaubq4le0e4gwlpiduqexfxq91f14op20ebg",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 375,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:31095/bucket-05f1jl9fqagrykcwo2uaaubq4le0e4gwlpiduqexfxq91f14op20ebg: dial tcp 72.28.97.59:31095: connect: connection refused",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```
<a name="431a59f9-39bf-4c6e-aba4-7973f8dcea38"></a>
### 7556-f913abb/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31172
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ef1ba49c0fc6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 1 minutes and 50 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 30 seconds
(8/14) Running minio-dotnet tests ... done in 35 seconds
(9/14) Running minio-go tests ... done in 53 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 32 seconds
(11/14) Running minio-js tests ... done in 38 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 9 seconds
{
  "args": {
    "bucket_name": "minio-py-test-a61302f8-53c3-4316-8c34-be99edb34e99",
    "object_name": "c21c232e-d2fc-49fd-b8e4-14de01427c2c"
  },
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1266, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 152, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1888, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1274, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "message": "'content-language'",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "status": "FAIL",
  "duration": 108
}

Executed 11 out of 14 tests successfully.
```

--- a/cmd/bucket-handlers.go
+++ b/cmd/bucket-handlers.go
@@ -774,10 +774,6 @@ func (api objectAPIHandlers) DeleteBucketHandler(w http.ResponseWriter, r *http.
 		return
 	}
 
-	globalNotificationSys.RemoveNotification(bucket)
-	globalPolicySys.Remove(bucket)
-	globalNotificationSys.DeleteBucket(ctx, bucket)
-
 	if globalDNSConfig != nil {
 		if err := globalDNSConfig.Delete(bucket); err != nil {
 			// Deleting DNS entry failed, attempt to create the bucket again.
@@ -786,6 +782,10 @@ func (api objectAPIHandlers) DeleteBucketHandler(w http.ResponseWriter, r *http.
 			return
 		}
 	}
+
+	globalNotificationSys.RemoveNotification(bucket)
+	globalPolicySys.Remove(bucket)
+	globalNotificationSys.DeleteBucket(ctx, bucket)
 
 	// Write success response.
 	writeSuccessNoContent(w)

--- a/pkg/dns/etcd_dns.go
+++ b/pkg/dns/etcd_dns.go
@@ -70,7 +70,20 @@ func (c *coreDNS) Get(bucket string) ([]SrvRecord, error) {
 		if err != nil {
 			return nil, err
 		}
-		srvRecords = append(srvRecords, records...)
+		// Make sure we have record.Key is empty
+		// this can only happen when record.Key
+		// has bucket entry with exact prefix
+		// match any record.Key which do not
+		// match the prefixes we skip them.
+		for _, record := range records {
+			if record.Key != "" {
+				continue
+			}
+			srvRecords = append(srvRecords, records...)
+		}
+	}
+	if len(srvRecords) == 0 {
+		return nil, ErrNoEntriesFound
 	}
 	return srvRecords, nil
 }

